File: 0045_med_hot_jump-game-ii/solution.java
class Solution {
    public int jump(int[] nums) {
        if (nums == null || nums.length <= 1) {
            return 0;
        }

        int curDistance = 0;
        int ans = 0;
        int nextDistance = 0;

        for (int i = 0; i < nums.length; i++) {
            nextDistance = Math.max(nextDistance, i + nums[i]);
            if (i == curDistance) {
                ans += 1;
                curDistance = nextDistance;
                if (nextDistance >= nums.length - 1) {
                    break;
                }
            }
        }

        return ans;

    }
}

File: 0045_med_hot_jump-game-ii/solution.py
from typing import List


class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)

        if n <= 1:
            return 0
        
        curDistance = 0   # 当前覆盖最远距离下标
        ans = 0           # 记录走的最大步数
        nextDistance = 0  # 下一步覆盖最远距离下标

        for i in range(n):
            nextDistance = max(nextDistance, i + nums[i]) # 更新下一步覆盖最远距离下标
            if i == curDistance:           # 遇到当前覆盖最远距离下标
                ans += 1                   # 需要走下一步
                curDistance = nextDistance # 更新当前覆盖最远距离下标（相当于加油了）
                if nextDistance >= n - 1:  # 当前覆盖最远距到达集合终点，不用做ans++操作了，直接结束
                    break
        return ans
            